deepr/io/path.py

```diff
@@ -131,6 +131,18 @@ def mkdir(self, parents: bool = False, exist_ok: bool = False, filesystem: FileS
             else:
                 pathlib.Path(str(self)).mkdir(parents=parents, exist_ok=exist_ok)
 
+    def delete_dir(self, filesystem: FileSystem = None):
+        """Delete dir from filesystem"""
+        if not self.is_dir(filesystem=filesystem):
+            raise FileNotFoundError(str(self))
+        if filesystem is not None:
+            filesystem.rm(str(self), recursive=True)
+        else:
+            if self.is_hdfs:
+                tf.io.gfile.rmtree(str(self))
```

Comment:
more consistent to do `with HDFSFileSystem() as hdfs:`

Reply:
done